process: describe Xen Troops development process

lorc opened the pull request:
This document is aimed to be our set of rules, which we will strictly follow. It should be considered as the final truth in any argument.

It describes important topics like "definition of done", licensing, submitting changes and so on.

lorc committed:
process: describe Xen Troops development process

This document is aimed to be our set of rules, which we will strictly
follow. It should be considered as the final truth in any argument.

It describes important topics like "definition of done", licensing,
submitting changes and so on.

Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Reviewed-by: Artem Mygaiev <artem_mygaiev@epam.com>
Reviewed-by: Ruslan Shymkevych <ruslan_shymkevych@epam.com>

rshym commented on process/process.md:
```diff
@@ -0,0 +1,295 @@
+
+# Definition of Done
+
+Every task can is considered completed when resulting changes are
```
> can is 
Seems like misprint.

rshym commented on process/process.md:
```diff
@@ -0,0 +1,295 @@
+## Coding style
+
+Every project has own coding style so we can't provide here any
+specific requirements. But you always should to adhere to the coding
```
s/"always should to adhere"/"should always adhere"

rshym commented on process/process.md:
````diff
@@ -0,0 +1,295 @@
+```
+subsystem: add possibility to grok baz
+
+When workin on foo, we noticed that blah-blah-blah.
````
s/workin/working

rshym commented on process/process.md:
```diff
@@ -0,0 +1,295 @@
+name, detailed description of a problem solved. Do not try to squash
+your commit messages, write description for the PR as one entity instead.
+
+If appropriate, write how you tested it. Wait automated checks to run.
```
Seems like s/"Wait automated"/"Wait for automated"

rshym commented on process/process.md:
```diff
@@ -0,0 +1,295 @@
+
+Every project has maintainer(s). Maintainers have full vision of
+project and its architecture. In most cases maintainers will support
+code you contributed for rest of it life. So maintainers can ask you
```
s/it/its

rshym commented on process/process.md:
```diff
@@ -0,0 +1,295 @@
+project and its architecture. In most cases maintainers will support
+code you contributed for rest of it life. So maintainers can ask you
+to make changes into your PR so it better fit into project's
+architecture. Even if your code is 100% and bug-free, it does not mean
```
> is 100% and bug-free
Seems something is missed here after "100%"

rshym commented on process/process.md:
```diff
@@ -0,0 +1,295 @@
+
+In some cases it is tempting to make temporary fix to get project
+going forward. But practice shows that this temporary fixes tend to
+stick forever. So we trying to avoid them at all costs. But sometimes
```
s/"we trying"/"we are trying"

rshym commented on process/process.md:
```diff
@@ -0,0 +1,295 @@
+   should look like "Rework temporary fix introduced by PR
+   #NNN". Assign yourself to it. Set resolution date. Make sure that
+   PR page has reference to this issue.
+
+
+
+
```
Not needed empty lines.

lorc commented:
Thanks @rshym, I have addressed your comments

rshym commented:
`Reviewed-by: Ruslan Shymkevych <ruslan_shymkevych@epam.com>`

klogg commented on process/process.md:
```diff
@@ -0,0 +1,291 @@
+tasks where you research some topic or experiment with different
+approaches. Those still require some tangible results, like Wiki page
+for example.
+
```
Probably worth to mention that it is up to committers / maintainers to ensure code quality, test coverage, documentation completeness prior to merging the PR(s)

klogg commented on process/process.md:
```diff
@@ -0,0 +1,291 @@
+
+# Licensing
+
+All new code should be licensed under Apache-2.0 license by default.
```
"New code" above means completely new projects (new app, new lib, new daemon, etc.)
New code to existing projects (i.e. new modules for Xen, Linux kernel, etc.) shall follow the license of existing projects.
Use SPDX tags where possible: spdx.github.io/spdx-spec/v3.0/annexes/using-SPDX-short-identifiers-in-source-files/
For new files and significantly changed (new major features, major refactoring, etc.) add EPAM's copyright line in a suitable format.
If usure on licensing/copyright, ask Artem.

klogg commented on process/process.md:
```diff
@@ -0,0 +1,291 @@
+
+All new code should be licensed under Apache-2.0 license by default.
+
+# "Temporary" changes
```
Better to move this section above "Check list" and have the "Issue created for fixing temporary change" item added to the check list

klogg commented on process/process.md:
```diff
@@ -0,0 +1,291 @@
+After collecting all needed R-b tags, you should include them
+into appropriate commit messages, rebase changes onto current upstream
+branch and force push changes into your local branch.
+
```
I miss here rules to update pull request when addressing reviewer's comments. Somehow our Renesas colleagues had issue with this in the past: https://github.com/aoscloud/meta-aos-rcar-gen5/pull/2#issuecomment-1959032876

klogg commented on process/process.md:
```diff
@@ -0,0 +1,291 @@
+# Preface
+
+This document defines processes adopted by Xen-Troops project. It
```
Only "introducing code changes" processes. Other processes we have not yet developed (i.e. task breakdown/estimation, triage, ...)

klogg commented on process/process.md:
```diff
@@ -0,0 +1,291 @@
+specific requirements. But you should always adhere to the coding
+style of a project where you making commit. When in doubt - look at
+surrounding code and follow its style.
+
```
Please suggest style to stick to when starting **new project(s)** in C, C++, Rust, Python, Bash. I really hate seeing CaMeLcAsE one day.

klogg commented on process/process.md:
```diff
@@ -0,0 +1,291 @@
+If you are adding public interfaces - document them in the same way,
+as other parts are documented or better. Always make sure that
+comment reflect the code.
+
```
Maybe start using Doxygen tags properly?

klogg commented on process/process.md:
```diff
@@ -0,0 +1,291 @@
+integrated into the codebase. This means that you have done your task
+only **after** your Pull Request(s) was/were merged. Before that
+moment your task is **not** done.
+
```
Sometimes DoD for us means code is in upstream. I suggest having a section on how to upstream (including mess with using private emails but submitting on behalf work emails)

lorc commented:
Thanks, @klogg 

I tried to address your comments. Could you please take a view?

For ease of review I added a fix-up patch with new changes.

Also, I added this paragraph, based on today's discussion:

> Try to avoid unsolicited changes during review phase. Do not add new
features or change existing code unless asked by reviewer. This will
increase review time drastically, as reviewers will need to review the
PR after any such change.

klogg commented:
`Reviewed-by: Artem Mygaiev <artem_mygaiev@epam.com>`
